Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -3074,6 +3086,15 @@ extension BrowserViewController: PreferencesObserver {
             setupTabs()
             updateTabsBarVisibility()
             updateApplicationShortcuts()
+        case Preferences.General.alwaysRequestDesktopSite.key:
+            tabManager.reset()
+            self.tabManager.reloadSelectedTab()
+            for tab in self.tabManager.allTabs where tab != self.tabManager.selectedTab {
```

Comment:
See: https://github.com/brave/brave-ios/pull/1368#discussion_r316375423

I think all of this logic is overkill. Let's run some tests without all of this looping logic (e.g. tab manager reset, and reloading selected tab). The consequence to this _should_ be lazy loaded alternative pages (e.g. when a user clicks on a new tab, it will be freshly loaded).

Not all tabs have webviews, so this could be a pretty bloated operation. Notice how `reset()` only resets tabs that have active webviews?

Reply:
Fixed.